client/src/components/Register/RegisterContainer.tsx

```diff
@@ -1,73 +1,28 @@
 import React from "react";
-import { connect } from "react-redux";
 import Register from "./Register";
-import {
-  setUsername,
-  setFullname,
-  setEmail,
-  setPassword,
-  registerUser
-} from "../../store/register/actions";
+import { reduxForm } from "redux-form";
+import API from "../../store/api";
+import { showAlert } from "../../store/alert/actions";
 
-interface FormProps {
-  username: string;
-  fullname: string;
-  email: string;
-  password: string;
-  setUsername: Function;
-  setFullname: Function;
-  setEmail: Function;
-  setPassword: Function;
-  registerUser: Function;
-}
-
-interface RegisterState {
-  username: string;
-  fullname: string;
-  email: string;
-  password: string;
-}
-
-interface FormState {
-  register: RegisterState;
-}
+class RegisterContainer extends React.Component<any> {
+  onSubmit = (user: any, dispatch: Function) => {
+    return API.post("/user", user)
```

Comment:
move to action and use props, don't forgot about pending, success, and error handling